internal/credential/service_list_stores.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package credential
+
+import (
+	"context"
+	"slices"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/pagination"
+	"github.com/hashicorp/boundary/internal/util"
+)
+
+// SubtypeStoreService defines the interface expected
+// to gather information about credential stores.
+type SubtypeStoreService interface {
+	EstimatedStoreCount(context.Context) (int, error)
+	ListDeletedStoreIds(context.Context, time.Time, ...Option) ([]string, error)
+	ListCredentialStores(context.Context, []string, ...Option) ([]Store, error)
+}
+
+// StoreService coordinates calls across different subtype services
+// to gather information about all credential stores.
+type StoreService struct {
+	services []SubtypeStoreService
+	writer   db.Writer
+}
+
+// NewStoreService returns a new credential store service.
+func NewStoreService(ctx context.Context, writer db.Writer, vaultService SubtypeStoreService, staticService SubtypeStoreService) (*StoreService, error) {
```

Comment:
In theory we could take a slice of services, but I think it's more error prone than having two explicit parameters.